File: backend/utils/loginValidator.js
// Уніфікована валідація логіну
const validateLogin = (login) => {
  const loginRegex = /^[a-zA-Z0-9_]{3,30}$/;
  
  if (!login) {
    return { isValid: false, message: 'Логін обов\'язковий' };
  }
  
  if (login.length < 3) {
    return { isValid: false, message: 'Логін має містити мінімум 3 символи' };
  }
  
  if (login.length > 30) {
    return { isValid: false, message: 'Логін має містити максимум 30 символів' };
  }
  
  if (!loginRegex.test(login)) {
    return { isValid: false, message: 'Логін може містити тільки латинські літери, цифри та підкреслення' };
  }
  
  return { isValid: true };
};

module.exports = validateLogin;

File: mobile/lib/core/validators/login_validator.dart
class LoginValidator {
  // Уніфікована валідація логіну (така ж як на backend)
  static ValidationResult validate(String? login) {
    if (login == null || login.isEmpty) {
      return ValidationResult(
        isValid: false,
        message: 'Логін обов\'язковий',
      );
    }

    final trimmedLogin = login.trim();

    if (trimmedLogin.length < 3) {
      return ValidationResult(
        isValid: false,
        message: 'Логін має містити мінімум 3 символи',
      );
    }

    if (trimmedLogin.length > 30) {
      return ValidationResult(
        isValid: false,
        message: 'Логін має містити максимум 30 символів',
      );
    }

    final loginRegex = RegExp(r'^[a-zA-Z0-9_]{3,30}$');
    if (!loginRegex.hasMatch(trimmedLogin)) {
      return ValidationResult(
        isValid: false,
        message: 'Логін може містити тільки латинські літери, цифри та підкреслення',
      );
    }

    return ValidationResult(isValid: true);
  }
}

class ValidationResult {
  final bool isValid;
  final String? message;

  ValidationResult({
    required this.isValid,
    this.message,
  });
}

